packages/protocol/contracts/L1/v1/V1Proving.sol

```diff
@@ -263,9 +273,17 @@ library V1Proving {
             fc.provenAt = uint64(block.timestamp);
         } else {
             require(
-                fc.blockHash == blockHash && fc.proposedAt == target.timestamp,
-                "L1:proof:conflict"
+                fc.proposedAt == target.timestamp,
+                "L1:proposedAt:conflict"
             );
+
+            if (fc.blockHash != blockHash) {
+                // We have a problem here: two proofs are both valid but claims
+                // the new block has different hashes.
+                V1Utils.halt(s, true);
+                return;
+            }
```

Comment:
If this also used as a potential fraud proof system (for the case where a circuit bug allows generating a proof for multiple block hashes) it seems like a good idea to add an artificial delay to the onchain finalization of a block (the fraud proof period) to allow good actors the time to submit this proof (otherwise it can already be too late). This delay will also be helpful for other methods of overriding the validity of a block (governance, or even full blown optimistic style fraud proofs).

Reply:
Added a verification delay of 60 minutes for now.  In the future, we should use a value calculated based on the average proof time. A TODO added.